openpype/hosts/maya/plugins/publish/extract_look.py

```diff
@@ -535,6 +535,9 @@ def _process_texture(self, filepath, do_maketx, staging, linearize, force):
                 self.log.info("tx: converting sRGB -> linear")
                 additional_args.extend(["--colorconvert", "sRGB", "linear"])
 
+            self.log.info("Using nuke-default ocio config instead of maya ocio config!") # noqa
+            self.log.info("The tx conversion is different from the maya tx conversion!") # noqa
```

Comment:
But why? Does it need to be? Couldn't it just use the OCIO path currently set in maya? Or?

Reply:
i have tested and written to set up custom ocio config here. but it seems that if the users use aces_1.2 ocio.config, the args for` --colorconvert `needs to be also adjusted. While you are using maya-defaults, the args may also need some changes. I am not sure if the args will vary by version too. It definitely needs time to research. So that's why I set it info here first and the config itself.

Reply:
Yep, this is because we know that in nuke-default there is `sRGB` and `linear`. But without parsing and getting proper color spaces from ocio config we can't auto-determine what to use for linearizations.

Reply:
I still think we're doing something wrong with our logic here. Arnold's built-in TX manager basically runs this over all images: `maketx -u -v --checknan --stats --unpremult --oiio %s`.

I've never really heard anyone complain about the results of that functionality. What's the reason we're instead trying to color convert it explicitly by default?

Seems like if _they_ (Autodesk/Arnold tools) don't do it by default then maybe we shouldn't either? And if we'd want to color convert (e.g. linearize) that maybe that logic should be achieved through settings?

Reply:
I don't have much Color Space knowledge myself, but e.g. here they mention that if by chance an input image is 8bit to just keep the color values as they are: https://forum.isotropix.com/viewtopic.php?f=5&p=25452

Of course a logical issue if the input is 8 bit - but maybe we shouldn't assume that our texture inputs aren't 8 bit either?

Reply:
> I still think we're doing something wrong with our logic here. Arnold's built-in TX manager basically runs this over all images: `maketx -u -v --checknan --stats --unpremult --oiio %s`.
> 
> I've never really heard anyone complain about the results of that functionality. What's the reason we're instead trying to color convert it explicitly by default?
> 
> Seems like if _they_ (Autodesk/Arnold tools) don't do it by default then maybe we shouldn't either? And if we'd want to color convert (e.g. linearize) that maybe that logic should be achieved through settings?

i have double checked the conversion again with maketx.exe and come up with the conclusion. The config file required for the --colorconvert command, otherwises it will error out.
![image](https://user-images.githubusercontent.com/64118225/212725726-d2060a4f-c807-4e07-afd4-5e952564c8bd.png)
I have practiced three different occassions with the maketx.exe in oiio and put them back into maya with Arnold renderer, finding out that the tx files without the color conversion along with the maya default ocio config, it will result different colors. The screenshot of the ipr render in Arnold kinda explains the case.
The sphere in the middle is with the shader where the tx file is autogenerated by Maya.The sphere in the left is with the shader where the albedo without` --colorconvert `args and the maya-defaults ocio config files added into args, i.e. `--colorconfig`.  The sphere in the right is with the shader where the albedo with both colorcovert args and maya-defaults ocio config set in the args during the tx conversion in maketx.exe. 
The color difference can also be seen when we are using different ocio config file, i.e. nuke-defaults.
![image](https://user-images.githubusercontent.com/64118225/212725321-1f56104d-daf4-4cdc-8959-d12df8869804.png)

Reply:
If I don't get it wrong, file texture node in Maya automatically set the colospace to Raw if the image format is tx. The sphere in the far left is the shader with albedo texture in tx format,which the tx file is generated without both color convert and color config args. The sphere in the middle is with the shader which the albedo using autogenerated tx file by Maya. And you can see the color difference in-between. The sphere in the right side is using tx map as albedo which is generated by both color convert and Maya-default ocio color config.

Reply:
Had a quick look on my side. Some notes:

- Whenever you directly load a `.tx` file into maya and render with arnold any color space settings you set on the `file` node do nothing whatsover. It seems Arnold will then always assume the texture itself is in the raw color space.
- The Maya viewport will change when you set the color space on a file node pointing to a `.tx` file, but arnold doesn't care at all. It takes in the `.tx` file as if it is raw.
- The auto-tx conversion does work.
- The TX manager conversion for me crashes in the latest version of arnold, yay. 

Tested in Maya 2022 + Mtoa 5.2.1.1

Might need to delve in deeper to really be able to help out. :)

Reply:
So here's a function of one of the scripts that comes with Arnold, `scripts/mtoa/makeTx.py`:
_(There was no license in the file, but I suppose copyright Autodesk?)_

<details>
  <summary>Expand to see ``scripts/mtoa/makeTx.py` from Mtoa 5.2.2.1</summary>
```python
def makeTx(filename, colorspace='auto', arguments=''):
    '''Generate a TX texture with maketx
    '''

    # status[0] contains the amount of created tx files
    # status[1] the amount of skipped tx files
    # status[2] the amount of errors

    status = [0,0,0]
    if arguments == '':
        cmd = list(_maketx_cmd)
    else:
        cmd_str = _maketx_binary
        cmd_str += ' '
        cmd_str += arguments
        cmd = shlex.split(cmd_str, posix=False)

    #maya_version = utils.getMayaVersion()
    if mel.eval("exists \"colorManagementPrefs\""):

        if cmds.colorManagementPrefs(q=True, cmEnabled=True):
            if colorspace in cmds.colorManagementPrefs(q=True, inputSpaceNames=True):
                if cmds.colorManagementPrefs(q=True, cmConfigFileEnabled=True):
                    color_config = cmds.colorManagementPrefs(q=True, configFilePath=True)
                else:
                    color_config = cmds.internalVar(userPrefDir=True)

                render_colorspace = cmds.colorManagementPrefs(q=True, renderingSpaceName=True)
                
                if colorspace != render_colorspace:
                    cmd += ['--colorengine', 'syncolor', '--colorconfig', color_config, '--colorconvert', colorspace, render_colorspace]
            else:
                # FIXME what should we do in auto mode ?
                if colorspace != 'auto':
                    print('[maketx] Warning: Invalid input colorspace "%s" for "%s", disabling color conversion' % (colorspace, filename))

    for tile in expandFilename(filename):
        if os.path.splitext(tile)[1] == '.tx':
            print('[maketx] Skipping native TX texture: %s' % tile)
            status[1] += 1
            continue

        tile_info = imageInfo(tile)

        if colorspace == 'auto':
            colorspace = guessColorspace(tile_info)

        outputTx = os.path.splitext(tile)[0] + '.tx'
        AiTextureInvalidate(outputTx)

        AiMsgInfo( "[maketx] info : {}".format(','.join(["{}={}".format(k,v) for k, v in tile_info.items()])) )

        # promote 8-bit images to EXR half with DWAA compression to avoid quantization errors (#3919)
        if colorspace != 'linear' and tile_info['bit_depth'] <= 8:
            cmd += ['--format', 'exr', '-d', 'half', '--compression', 'dwaa']

        res = subprocess.Popen(cmd + [tile], stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.PIPE, startupinfo=_no_window).communicate()[0]

        if re.search(_maketx_rx_noupdate, res):
            # we don't want to get messages saying that texture is up-to-date, mainly
            # with auto-tx since this happens at every render.
            # If we really want it for txManager we can add an argument in this function
            # and re-introduce the print

            #print '[maketx] TX texture is up to date for "%s" (%s)' % (tile, colorspace)
            status[1] += 1
        else:
            mo = re.search(_maketx_rx_stats, res)
            if mo:
                print('[maketx] Generated TX for "%s" (%s) in %s seconds' % (tile, colorspace, mo.group(1)))
                AiTextureInvalidate(outputTx) 
                status[0] += 1
            else:
                print('[maketx] Error: Could not generate TX for "%s" (%s)' % (tile, colorspace))
                print(res)
                status[2] += 1
       
    return status
```
</details>

This has some nice information, like the 8 bit conversion they are doing, etc.

There's also the `scripts/mtoa/txManager/lib.py` which could be a good reference to check too?

### The Arnold Tx manager conversion

The TX Manager actually does the conversion from "node space" to "render space".
They do indeed do `--colorconvert`.

This is the Render Space:
```python
render_colorspace = cmds.colorManagementPrefs(query=True, renderingSpaceName=True)
```
And the texture's color space is:
```
color_space = cmds.getAttr(file_node + ".colorSpace")
```

As such, looking at that. The logic we should implement for maketx is to whenever Color Management is enabled to take the Color Space of the texture an convert it to the render space.

Et voila, issue solved.

TL;DR

For each node we should do:
The current node's color space to the render color space, that's it.

Reply:
> @antirotor @moonyuet With above comment I feel we can solve this PR easily, yes? :) We'll just always need to do the color convert with the currently used OCIO config in maya and convert from the node's color space to the rendering space.

I am not quite sure if the "nuke-defaults" ocio config is set for some purpose in the code and let's assume yes. It would be great if there is an option to use the maya_defaults ocio config for tx conversion if it turns out it will give an accurate tx conversion.

Reply:
> It would be great if there is an option to use the maya_defaults ocio config for tx conversion if it turns out it will give an accurate tx conversion.

Just to be exactly clear: I'm not saying to use Maya's default - but to use the config that maya is currently set to on the artist's machine. We can make the assumption that if color space management is enabled for Maya within OpenPype that OpenPype sets up Maya to be in the correct OCIO state.